airbyte-integrations/connectors/source-github/unit_tests/test_source.py

```diff
@@ -138,30 +137,63 @@ def test_organization_or_repo_available():
     assert exc_info.value.args[0] == "No streams available. Please check permissions"
 
 
-@pytest.mark.parametrize(
-    ("repos_config", "expected"),
-    [
-        (("airbytehq/airbyte/", "airbytehq/", "airbytehq", "airbyte hq", "airbytehq/*/", "adc/ff*f", "akj*/jahsd"), False),
-        (("airbytehq/airbyte", ), True),
-        (("airbytehq/airbyte-test", "airbytehq/airbyte_test", "airbytehq/airbyte-test/another-repo"), True),
-        (("air232bytehq/air32byte", "airbyte_hq/another-repo", "airbytehq/*", "airbytehq/airbyte"), True),
-        (("airbyte_hq/another.repo", "airbytehq/*", "airbytehq/airbyte"), True),
-    ],
-)
-def test_config_validation(repos_config, expected):
-    actual = SourceGithub._is_repositories_config_valid(repos_config)
-    assert actual == expected
-
-
-@pytest.mark.parametrize(
-    ("config", "expected"),
-    [
-        ({"access_token": "test_token", "repository": "airbytehq/airbyte-test"}, {"airbytehq/airbyte-test", }),
-        ({"access_token": "test_token", "repository": "airbytehq/airbyte-test "}, {"airbytehq/airbyte-test", }),
-        ({"access_token": "test_token", "repository": "airbytehq/airbyte-test/"}, {"airbytehq/airbyte-test", }),
-        ({"access_token": "test_token", "repository": "airbytehq/airbyte-test/  airbytehq/airbyte"}, {"airbytehq/airbyte", "airbytehq/airbyte-test"}),
-    ],
-)
-def tests_get_and_prepare_repositories_config(config, expected):
-    actual = SourceGithub._get_and_prepare_repositories_config(config)
-    assert actual == expected
+def tests_get_and_prepare_repositories_config():
+    config = {"repository": "airbytehq/airbyte airbytehq/airbyte-test  airbytehq/integration-test.git   airbytehq/airbyte.git"}
+    assert SourceGithub._get_and_prepare_repositories_config(config) == {"airbytehq/airbyte", "airbytehq/airbyte-test", "airbytehq/integration-test"}
+
+
+def test_check_config_repository():
+    source = SourceGithub()
+    source.check = MagicMock(return_value=True)
+    config = {"credentials": {"access_token": "access_token"}, "start_date": "1900-01-01T00:00:00Z"}
+
+    repos_ok = [
+        "airbytehq/airbyte",
+        "airbytehq/airbyte-test",
+        "airbytehq/airbyte_test",
+        "airbytehq/airbyte.git",
+        "airbytehq/*",
+        "airbytehq/.",
+        "airbyte_hq/airbyte",
+        "airbytehq/123",
+    ]
+
+    repos_fail = [
+        "airbytehq",
+        "airbytehq/",
+        "airbytehq/*/",
+        "airbytehq/airbyte/",
+        "airbytehq/air*yte",
+        "airbyte*/airbyte",
+        "airbytehq/airbyte-test/master-branch",
+        "https://github.com/airbytehq/airbyte",
+    ]
```

Comment:
1. This case `https://github.com/openai/gpt-3.git` - Fails because of the prefix "https://github.com/"
2. But for example, such case: `openai/gpt-3.git`  - will be MATCHED by regex but at later stage normalized to -> `openai/gpt-3`

Reply:
Why'd we decide to do them differently- can we also just not let them end in .git in the pattern?

Reply:
I have re-implemented, now it's forbidden to enter ending '.git' too

Reply:
sweet 👌🏻